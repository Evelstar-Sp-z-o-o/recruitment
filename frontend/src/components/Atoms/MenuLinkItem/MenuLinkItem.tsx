import { Link } from 'react-router-dom';

import ListItemText from '@mui/material/ListItemText';
import MenuItem from '@mui/material/MenuItem';

interface IMenuLinkItemProps {
  target?: string;
  label: string;
}

const MenuLinkItem: FC<IMenuLinkItemProps> = ({ target, label }) => {
  return (
    <MenuItem>
      <Link to={target} className="menuItem">
        <ListItemText>{label}</ListItemText>
      </Link>
    </MenuItem>
  );
};

export default MenuLinkItem;
