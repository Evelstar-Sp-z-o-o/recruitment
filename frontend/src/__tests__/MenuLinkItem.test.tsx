import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';

import MenuLinkItem from '@/src/components/Atoms/MenuLinkItem/MenuLinkItem';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

describe('MenuLinkItem Component', () => {
  it('renders MenuLinkItem component correctly', () => {
    render(
      <MemoryRouter>
        <MenuLinkItem target="/home" label="Home" />
      </MemoryRouter>,
    );

    const menuItem = screen.getByRole('menuitem');
    expect(menuItem).toBeInTheDocument();

    const link = screen.getByRole('link', { name: /Home/i });
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute('href', '/home');
    expect(link).toHaveClass('menuItem');

    const listItemText = screen.getByText('Home');
    expect(listItemText).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const tree = create(
      <MemoryRouter>
        <MenuLinkItem target="/home" label="Home" />
      </MemoryRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
