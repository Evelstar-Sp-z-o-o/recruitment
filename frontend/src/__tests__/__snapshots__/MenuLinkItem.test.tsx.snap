// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MenuLinkItem Component > should match snapshot 1`] = `
<li
  className="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="menuitem"
  tabIndex={-1}
>
  <a
    className="menuItem"
    href="/home"
    onClick={[Function]}
  >
    <div
      className="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Home
      </span>
    </div>
  </a>
</li>
`;
